MergerLogic/ImageProcessing/TileScaler.cs

```diff
@@ -9,25 +9,52 @@ public class TileScaler : ITileScaler
         private const int TILE_WIDTH = 256;
         private const int TILE_HEIGHT = 256;
 
-        public void Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
+        public MagickImage Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
         {
             int zoomLevelDiff = targetCoords.Z - baseTile.Z;
             int scale = 1 << zoomLevelDiff;
-            double scaleAsDouble = (double)scale;
 
-            double tilePartX = (targetCoords.X % scale) / scaleAsDouble;
-            double tilePartY = (targetCoords.Y % scale) / scaleAsDouble;
-            double tileSize = TILE_HEIGHT / scaleAsDouble;
+            //find source pixels range
+            double tilePartX = targetCoords.X % scale;
```

Comment:
Why did you remove the part dividing by the scale?
The scale is not absolute, it is the difference in scales, so it should be divided by.

Reply:
it want removed it was optimized.
the old calculation divided it by scale and then multiplied it by the base tile size(256)
now it is multiplied by the sub tile size instead (256/scale) so the calculation wasn't changed but less operations are needed